### ISSUE TITLE
Cannot install

### ISSUE THREAD

**Issue by llvilanova** [proofit404/anaconda-mode#181]
Hi! I found a couple problems during installation (`anaconda-mode-install`), and was a bit baffled until I realized there's an `*anaconda-mode*` buffer with useful information. Maybe that should be hinted as an error when installation (or something else) fails.

The first problem is that the pip module was not installed on my system. I have the virtualenv command installed instead (which in turn will install the relevant pip module). Migrating to it might be beneficial to keep installations self-contained (e.g., `virtualenv --system-site-packages /path/to/dir`).

The second problem happens during the installation itself:

<pre>
Installing collected packages: jedi, six, service-factory, anaconda-mode
[31mException:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 209, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 335, in run
    prefix=options.prefix_path,
  File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 732, in install
    **kwargs
  File "/usr/lib/python2.7/dist-packages/pip/req/req_install.py", line 835, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/lib/python2.7/dist-packages/pip/req/req_install.py", line 1030, in move_wheel_files
    isolated=self.isolated,
  File "/usr/lib/python2.7/dist-packages/pip/wheel.py", line 247, in move_wheel_files
    prefix=prefix,
  File "/usr/lib/python2.7/dist-packages/pip/locations.py", line 153, in distutils_scheme
    i.finalize_options()
  File "/usr/lib/python2.7/distutils/command/install.py", line 289, in finalize_options
    raise DistutilsOptionError("can't combine user with prefix, "
DistutilsOptionError: can't combine user with prefix, exec_prefix/home, or install_(plat)base[0m
[33mYou are using pip version 8.1.1, however version 8.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.[0m
[0;31mOut[[1;31m1[0;31m]: [0m2
[TerminalIPythonApp] WARNING | File not found: u'anaconda_mode.py'
</pre>

Thanks,
  Lluis

**Comment by proofit404** [comment]
Hello.

This covers few anaconda-mode problems:

* IPython exit with 0 exit code if there was an installation failure.
* Target installation fails again

**Comment by Nolski** [comment]
So what's the solution to this issue?

**Comment by proofit404** [comment]
This can't be solved in the simple way. We must find new workaround for IPython exit code (`os._exit` seems not work anymore). And finally workaround distutils bug which broke `-t` pip option.

**Comment by Nolski** [comment]
Then I'd argue that this issue shouldn't be closed if the issue cannot be solved or does not have a workaround.

**Comment by proofit404** [comment]
Consider it as duplicate. There is no reason to have two issues pointed to the same problem.

**Comment by Nolski** [comment]
Ah, my mistake! I thought this issue was closed as well. 😆

**Comment by shwsun** [comment]
@proofit404 I thought this is quite important but I dont think I need to create a issue so I just post a comment here.

The KeyError is quite the first and biggest issue, I believe somehow in the README "AttributeError and KeyError randomly happens" we should mention that user has to restart the Emacs or kill the anaconda-mode buffer to make anaconda functional.

**Comment by proofit404** [comment]
If you can improve README consider to make pull request.

**Comment by liweitianux** [comment]
Hello,

I just came across this installation issue after updating my ``anaconda-mode``, which was caused by the broken ``-t`` option of ``pip``, and it always complains:
```
DistutilsOptionError: can't combine user with prefix, exec_prefix/home, or install_(plat)base
```

After many searches, tries and failures, I finally figured this out.  And it turned out to be that the ``pip`` **shipped by Debian** (testing, amd64) just has issues with the ``-t`` option.  While the ``pip`` on my laptop (Gentoo, amd64, from portage) works well with the ``-t`` option, and there is no problem with installing ``anaconda_mode``.

Then, I tried to install another ``pip`` for my own just using the system ``pip``, and use the **newly installed** user ``pip`` to install ``anaconda_mode`` with the ``-t`` option.  It **just works**.

So, I have following two suggestions to this installation issue with broken ``pip`` is:

* Directly install another ``pip`` from pypi:
```
$ pip install --user pip
$ ~/.local/bin/pip install -t <target_dir> anaconda_mode
```
* Hack with the ``PYTHONUSERBASE`` environment variable:
```
$ mkdir <tmp_dir>
$ PYTHONUSERBASE=<tmp_dir> pip install --user anaconda_mode
$ mv <tmp_dir>/lib/python-2.7/site-packages/* <target_dir>
```

Hope this be helpful.

Cheers,
Aly
